Engine/Program.cs

```diff
@@ -0,0 +1,239 @@
+                SlowestResponseTimeMs = 5000
+            }
+        },
+        {
+            BotDifficulty.Medium,
+            new BotData
+            {
+                Name = "Harrowing Hayden", CustomIntroMessage = "He's a bit of a trickster so watch out",
+                CustomLoseMessage = "Damn, he's tricky", CustomWinMessage = "Down goes the trickster",
+                QuickestResponseTimeMs = 2000, SlowestResponseTimeMs = 4000
+            }
+        },
+        {
+            BotDifficulty.Hard,
+            new BotData
+            {
+                Name = "Masterful Mikaela", CustomIntroMessage = "She can't be trusted",
+                CustomLoseMessage = "Oof, rough one", CustomWinMessage = "Down falls Mikaela and her wicked ways",
+                QuickestResponseTimeMs = 1000,
+                SlowestResponseTimeMs = 2000
+            }
+        },
+        {
+            BotDifficulty.Impossible,
+            new BotData
+            {
+                Name = "Chaotic Kate", CustomIntroMessage = "rip lol", CustomLoseMessage = "No chance",
+                CustomWinMessage = "No one will ever see this message so it doesn't matter",
+                QuickestResponseTimeMs = 500,
+                SlowestResponseTimeMs = 1500
+            }
+        }
+    };
+
+    public static BotData Bot;
+
+    public static GameState gameState { get; set; }
+
+    public static void Main(string[] args)
+    {
+        PlayGame();
+
+        void PlayGame(bool skipIntro = false)
+        {
+            // Difficulty
+            BotDifficulty botDifficulty = CliGameUtils.GameIntro(skipIntro);
+            Bot = Bots[botDifficulty];
+            (int min, int max) botSpeedMs = (Bot.QuickestResponseTimeMs, Bot.SlowestResponseTimeMs);
+
+            gameState = GameEngine.NewGame(new List<string> {Bot.Name, "You"});
+            UpdateMessage("Game started!");
+
+            // Main game loop
+            while (GameEngine.TryGetWinner(gameState).Failure)
+            {
+                HandleUserInput(ReadlineWithBot());
+            }
+
+            // End game
+            Player winner = GameEngine.TryGetWinner(gameState).Data;
+            bool winnerIsPlayer = gameState.Players.IndexOf(winner) == 1;
+
+            Console.WriteLine("------ Game over ----");
+            Console.WriteLine();
+            Console.WriteLine(winnerIsPlayer ? Bot.CustomWinMessage : Bot.CustomLoseMessage);
+            Console.WriteLine();
+            Console.WriteLine();
+            Console.ForegroundColor = winnerIsPlayer ? ConsoleColor.DarkBlue : ConsoleColor.DarkRed;
+            UpdateMessage($"Winner is {winner.Name}");
+            Console.ForegroundColor = ConsoleColor.Black;
+            Console.WriteLine();
+            Console.WriteLine("Replay? (y/n)");
+            if (Console.ReadLine() == "y")
+            {
+                PlayGame(true);
+            }
+        }
+
+
+        void HandleUserInput(string? input)
```

Comment:
Similar to above I think alot of this command line logic could exist in its own class and the program just starts a game and exits when it finishes